File: batches_javascript/batch_10_code_reordering/compare.js
function computeStats(data) {
    const total = data.reduce((s, v) => s + v, 0);
    const avg = data.length > 0 ? total / data.length : 0;
    let mx = data[0], mn = data[0];
    for (const v of data) {
        if (v > mx) mx = v;
        if (v < mn) mn = v;
    }
    return { avg, max: mx, min: mn };
}

const d = [10, 20, 30, 40, 50];
console.log(computeStats(d));

File: batches/batches/batch_10_code_reordering/compare.py
"""
Comparison - Code Reordering (CHEATING)
Same logic, independent statements reordered.
"""


def compute_stats(data):
    n = len(data)
    mx = max(data)
    mn = min(data)
    total = sum(data)
    avg = total / n if n else 0
    return avg, mx, mn


if __name__ == "__main__":
    d = [10, 20, 30, 40, 50]
    print(compute_stats(d))
